packages/hardhat/contracts/FundingVault.sol

```diff
@@ -233,7 +259,25 @@ contract FundingVault{
         if (!tokenTransferSuccess){
             revert FundingVault__TokenTransferFailed();
         }
-        
+    }
+
+    /**
+     * @notice Get funding vault details
+     * @dev to access all neccassary parameters of the funding vault
+     */ 
+    function getVaults() external view returns(Vault memory)
```

Comment:
Instead of making all variables in lines 55--67 private and then wrapping them all into a struct to make them readable via this view function, make them all public. Then you cna read them directly.

Remove this function.